### ISSUE TITLE
Лучше *apply с передачей индексов 

### ISSUE THREAD

**Issue by gazpromneft-opti** [gazpromneft-opti/AutoOrder#18]
https://github.com/gazpromneft-opti/AutoOrder/blob/6b2b418bf4b161635ad73cb8bd8ae86141eaaf0d/PrepData.R#L23

**Comment by SergeyDan** [comment]
apply можно применять к конкретным элементам вектора?

**Comment by gazpromneft-opti** [comment]
Да, предварительно создав нужные индексы. Правда, стоит обратить внимание на оптимальность генерации индексов.

**Comment by SergeyDan** [comment]
есть вектор "y" c данными, есть вектор "ind"  с индексами тех значений, к которым нужно применить преобразование ниже.
  for (i in ind)
  { if (i<(q+1)) y[i]<-mean(y[(i+1):(i+1+2*q)])
    else if (i>(length(x)-q-1)) y[i]<-mean(y[(i-2*q-1):(i-1)])
    else y[i]<-mean(c(y[(i-q-1):(i-1)],y[(i+1):(i+q+1)]))
  }
правильно понимаю, что можно применить apply(y, ? , {...}, ...)

**Comment by SergeyDan** [comment]
tapply или sapply, думаю
